src/lib/modelUtils.ts

```diff
@@ -0,0 +1,335 @@
+ * @returns The remapped data.
+ */
+const remap = (
+  data: object | object[],
+  remappingDefinition: RemappingDefinition,
+  options:{
+    reverse?: boolean,
+    keepUnmappedValues?: boolean,
+    deleteMappedValues?: boolean,
+    deleteEmptyParents?: boolean,
+  } = {},
+): {
+  mapped: object | object[],
+  unmapped: object | object[],
+} => {
+  // If jsonData is null or undefined, return null
+  if (data === null || data === undefined) return null;
+  const {
+    reverse = false,
+    deleteMappedValues = true,
+    deleteEmptyParents = true,
+    keepUnmappedValues = true,
+  } = options;
+
+  let remappedData;
+  let unmappedData = data;
+  if (keepUnmappedValues) {
+    remappedData = data;
+  } else if (Array.isArray(data)) {
+    remappedData = [];
+  } else {
+    remappedData = {};
+  }
+
+  // Iterate over each key in the remapping definition
+  Object.keys(remappingDefinition).forEach((key) => {
+    // Determine the source and target paths based on the reverse flag
+    // eslint-disable-next-line no-nested-ternary
+    const sourcePath:string = reverse
+      ? Array.isArray(remappingDefinition[key])
+        ? remappingDefinition[key].slice(-1)
+        : remappingDefinition[key]
+      : key;
+    const targetDefinition = reverse
+      ? key
+      : remappingDefinition[key];
+    const targetActions = Array.isArray(targetDefinition)
+      ? targetDefinition
+      : [targetDefinition];
+    // Get the value from the source path in the remapped data
+    let sourceValue = dotWild.get(
+      keepUnmappedValues
+        ? remappedData
+        : data,
+      sourcePath,
+    );
+
+    // If the source value exists
+    if (sourceValue !== undefined) {
+      targetActions.forEach((targetAction) => {
+        if (targetAction instanceof Function) {
+          sourceValue = targetAction(sourceValue);
+          // Set the source value at the target path in the remapped data
+        } else if (Array.isArray(sourceValue) && targetAction.includes('*')) {
+          sourceValue.forEach((value, index) => {
+            const updatedTargetAction = targetAction.replace('*', index.toString());
+            remappedData = dotWild.set(remappedData, updatedTargetAction, value);
+          });
+        } else {
+          remappedData = dotWild.set(remappedData, targetAction, sourceValue);
+        }
+      });
+
+      // if keepUnmappedValues && deleteMappedValues, remove sourcePath and empty parent structures
+      if (keepUnmappedValues && deleteMappedValues) {
+        remappedData = remapPrune(remappedData, sourcePath, { deleteEmptyParents });
+      }
+      unmappedData = remapPrune(unmappedData, sourcePath, { deleteEmptyParents });
+    }
+  });
+
+  return { mapped: remappedData, unmapped: unmappedData };
+};
+
+/**
+ * Checks if a value is an object.
+ *
+ * @param value - The value to be checked.
+ * @returns A boolean indicating whether the value is an object or not.
+ */
+const isObject = (value: any): boolean => (
```

Comment:
I believe this is equivalent to "isPlainObject" from lodash if we don't want to maintain our own (we're already using lodash all over the place, so it wouldn't be adding a dependency)